src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -167,7 +167,7 @@ pub fn update_lockfile(ws: &Workspace<'_>, opts: &UpdateOptions<'_>) -> CargoRes
                 format!(
                     "{} -> #{}",
                     removed[0],
-                    &added[0].source_id().precise_git_fragment().unwrap()
+                    &added[0].source_id().precise_git_fragment().unwrap()[..8],
```

Comment:
Just a side comment, to be pedantically correct this should probably be using [`Object::short_id`](https://docs.rs/git2/latest/git2/struct.Object.html#method.short_id) to avoid collisions, but that seems pretty unlikely.